omnikinverter/omnikinverter.py

```diff
@@ -76,24 +71,36 @@ async def request(
             "Accept": "text/html,application/xhtml+xml,application/xml",
         }
 
-        if self._session is None:
-            self._session = ClientSession()
-            self._close_session = True
-
         auth = None
         if self.username and self.password:
             auth = aiohttp.BasicAuth(self.username, self.password)
 
         try:
             with async_timeout.timeout(self.request_timeout):
-                response = await self._session.request(
+                async with aiohttp.request(
                     "GET",
                     url,
                     auth=auth,
                     params=params,
                     headers=headers,
-                )
-                response.raise_for_status()
+                ) as response:
+                    response.raise_for_status()
+
+                    types = [
+                        "application/json",
+                        "application/x-javascript",
+                        "text/html",
+                    ]
+                    content_type = response.headers.get("Content-Type", "")
+                    if not any(item in content_type for item in types):
+                        text = await response.text()
+                        raise OmnikInverterError(
+                            "Unexpected response from the Omnik Inverter device",
+                            {"Content-Type": content_type, "response": text},
+                        )
+
+                    return await response.text()
```

Comment:
Why is this placed here?

Reply:
When using the [aiohttp Basic API](https://docs.aiohttp.org/en/stable/client_reference.html#basic-api)  which is without a session it looks like it requires a `with` block. I was not able to use the `response` outside of the with (which makes sense).

So I moved the exsiting code that handles the response within the `with` block. The logic itself is still the same, just moved.

Reply:
Ahh oke 👍🏻